samples/navigate-route-with-rerouting/src/main/java/com/esri/arcgismaps/sample/navigateroutewithrerouting/components/NavigateRouteWithReroutingViewModel.kt

```diff
@@ -0,0 +1,400 @@
+                    routeTracker = routeTracker,
+                    locationDataSource = simulatedLocationDataSource
+                )
+                locationDisplay.apply {
+                    // Set the simulated location data source as the location data source for this app
+                    dataSource = routeTrackerLocationDataSource.also {
+                        // Start the location data source
+                        it.start().onFailure { return@launch messageDialogVM.showMessageDialog(it) }
+                    }
+                    // Set the auto pan to navigation mode
+                    setAutoPanMode(LocationDisplayAutoPanMode.Navigation)
+                }
+                // Plays the direction voice guidance
+                updateVoiceGuidance(routeTracker)
+                // Zoom in the scale to focus on the navigation route
+                mapViewProxy.setViewpointScale(10000.0)
+            }
+            launch {
+                // Listen for changes in location
+                locationDisplay.location.collect {
+                    // Get the route's tracking status
+                    val trackingStatus = routeTracker.trackingStatus.value ?: return@collect
+                    // Displays the remaining and traversed route
+                    updateRouteGraphics(trackingStatus)
+                    // Display route status and directions info
+                    displayRouteInfo(trackingStatus)
+                    // Disable navigation button
+                    isNavigateButtonEnabled = false
+                }
+            }
+            launch {
+                // Automatically enable recenter button when navigation pan is disabled
+                locationDisplay.autoPanMode.filter { it == LocationDisplayAutoPanMode.Off }
+                    .collect { isRecenterButtonEnabled = true }
+            }
+            launch {
+                routeTracker.rerouteStarted.collect {
+                    nextDirectionText = "Re-routing..."
+                }
+            }
+        }
+    }
+
+    /**
+     * Displays the route distance and time information using [trackingStatus].
+     * When destination is reached the location data source is stopped.
+     */
+    private suspend fun displayRouteInfo(trackingStatus: TrackingStatus) {
+        // Get remaining distance information
+        val remainingDistance = trackingStatus.destinationProgress.remainingDistance
+        // Convert remaining minutes to hours:minutes:seconds
+        val remainingTimeString = DateUtils.formatElapsedTime(
+            (trackingStatus.destinationProgress.remainingTime * 60).toLong()
+        )
+        // Update text views
+        timeRemainingText = getString(R.string.time_remaining) + " " + remainingTimeString
+        distanceRemainingText = getString(R.string.distance_remaining) + " " +
+                remainingDistance.displayText + " " +
+                remainingDistance.displayTextUnits.abbreviation
```

Comment:
i would suggest moving the titles, `time_remaining ` and `distance_remaining` to the `NavigationRouteInfo` composable so that you're able to apply different configuration to these. For example, make them bold and bigger font.
This way, the visibility of the route information can be improved